core/src/Http/HttpRequest.cs

```diff
@@ -74,24 +85,47 @@ private static HttpRequestMessage CreateRequestMessage(Uri endpoint, Dictionary<
             return requestMessage;
         }
 
+        private static async Task<HttpContent> DeepCopy(HttpContent content)
+        {
+            var temp = new System.IO.MemoryStream();
+            await content.CopyToAsync(temp).ConfigureAwait(false);
+            temp.Position = 0;
+            var clone = new StreamContent(temp);
+            if (content.Headers != null)
+            {
+                foreach (var h in content.Headers)
+                {
+                    clone.Headers.Add(h.Key, h.Value);
+                }
+            }
+            return clone;
+        }
```

Comment:
pls. move to separate extensions file

Reply:
Quoted from [an earlier conversation](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r201542733):

> `HttpContent` has a `MemberWiseClone()` doing the shallow copy, but [does not seem to have a deep copy](https://msdn.microsoft.com/en-us/library/system.net.http.httpcontent(v=vs.110).aspx#Anchor_4). My current implementation was inspired by [this StackOverflow answer](https://stackoverflow.com/a/34049029).

It is currently only used in this class, so I put it as a private helper here. I can move it to core/src/Helpers/CoreHelpers.cs in next commit. It is just a matter of preference. #ConsiderItDone?

Reply:
Now we moved this helper function to another helper file.

---
In reply to: [205346722](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r205346722) [](ancestors = 205346722)